configs/components/rubygem-net-netconf.rb

```diff
@@ -1,17 +1,35 @@
 component "rubygem-net-netconf" do |pkg, settings, platform|
   pkg.version "0.4.3"
-  pkg.md5sum "fa173b0965766a427d8692f6b31c85a4"
-  pkg.url "http://buildsources.delivery.puppetlabs.net/net-netconf-#{pkg.get_version}.gem"
+  if platform.is_huaweios?
+    # This tarball was generated by doing a puppet-agent build without
+    # net-netconf but including its dependencies (nokogiri), installing
+    # it on the target, and installing the net-netconf gem using our gem
+    # command. The added files from :gem_home were then tar'ed up and
+    # packed into this container tarball.
+    pkg.url "http://buildsources.delivery.puppetlabs.net/net-netconf-#{pkg.get_version}.tar.gz"
+    pkg.md5sum "73ba0faadf9f3e3fcc8d9307b0ba4fa2"
+  else
+    pkg.url "http://buildsources.delivery.puppetlabs.net/net-netconf-#{pkg.get_version}.gem"
+    pkg.md5sum "fa173b0965766a427d8692f6b31c85a4"
+  end
 
   pkg.build_requires "ruby"
   pkg.build_requires "rubygem-nokogiri"
   pkg.build_requires "rubygem-net-scp"
 
-  # Because we are cross-compiling on ppc, we can't use the rubygems we just built.
-  # Instead we use the host gem installation and override GEM_HOME. Yay?
-  pkg.environment "GEM_HOME" => settings[:gem_home]
```

Comment:
That was a valid catch, but I have a suggestion from Huas on a change I can make to the net-netconf installation process that will make that code relevant again. As soon as I get my facter PR merged, I will return to this issue and hope to eliminate the need for this custom net-netconf tarball.